src/main/java/org/jboss/windup/operator/controller/WindupController.java

```diff
@@ -0,0 +1,80 @@
+package org.jboss.windup.operator.controller;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.Watcher;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import lombok.extern.java.Log;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.windup.operator.model.WindupResource;
+import org.jboss.windup.operator.model.WindupResourceDoneable;
+import org.jboss.windup.operator.model.WindupResourceList;
+import org.jboss.windup.operator.util.WindupDeployment;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Log
+@ApplicationScoped
+public class WindupController implements Watcher<WindupResource> {
+	@Inject
+	MixedOperation<WindupResource, WindupResourceList, WindupResourceDoneable, Resource<WindupResource, WindupResourceDoneable>> crClient;
+
+	@Named("namespace")
+	String namespace;
+
+	@ConfigProperty(name = "operator.serviceaccount", defaultValue = "windup-operator")
+	String serviceAccount;
+
+	@ConfigProperty(name = "operator.sso_public_key")
+	String ssoPublicKey;
+
+	@Inject
+	KubernetesClient k8sClient;
+
+	private void onAdd(WindupResource resource) {
+		log.info("Event ADD " + resource.getMetadata().getName());
+		if (!resource.isDeploying() && !resource.isReady()) {
+			new WindupDeployment(resource, crClient, k8sClient, namespace, serviceAccount, ssoPublicKey).deploy();
+		}
+	}
+
+	private void onUpdate(WindupResource newResource) {
+		log.info("Event UPDATE " + newResource.getMetadata().getName() + " - DeploymentsReady "
+				+ newResource.deploymentsReady() + " isReady " + newResource.isReady() 
+				+ " Status " + newResource.getStatus().getConditions());
+
+		// Consolidate status of the CR
+		if (newResource.deploymentsReady() == 3 && !newResource.isReady()) {
+			newResource.setReady(true);
+			newResource.setStatusDeploy(false); 
+
+			log.info("Setting this CustomResource as Ready");
+			crClient.inNamespace(namespace).updateStatus(newResource);
+		}
+	}
+
+	private void onDelete(WindupResource resource) {
+		log.info("Event DELETE [" + resource + "]");
+	}
+
+	@Override
+	public void eventReceived(Action action, WindupResource resource) {
+		log.info("Event received " + action + " received for WindupResource : " + resource.getMetadata().getName());
+
+		if (action == Action.ADDED) onAdd(resource);
```

Comment:
_Comment to further review the code, NOT mandatory for this PR_
It seems like this is not managing the [`Action.ERROR`](https://github.com/fabric8io/kubernetes-client/blob/v4.12.0/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Watcher.java#L30) case.

Furthermore, we could use a "switch-case" block with this `Action` enum.

Reply:
At the moment the ERROR is not covered as in this phase the operator is not controlling when pods fail to reconcile them up or down.....
I haven't found much more information about the ERROR event to know exactly what covers
.....
I think it's related with the conection more than the resources, as we have the executor and the mta giving a health checks == false, and even executor restarts once, and I think we never receive an ERROR event
so, seems more related to the conection , I would suggest to consider this as part as a total reconciliation when we cover the number of pods request feature
Also investigate how to test it

Reply:
+1 to the switch-case